dotcom-rendering/src/web/components/EnhancePinnedPost.importable.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react';
+
+const isServer = typeof window === 'undefined';
+
+const pinnedPost: Element | null = !isServer
+	? window.document.querySelector('[data-gu-marker=pinned-post]')
+	: null;
+
+const pinnedPostCheckBox: Element | null = !isServer
+	? window.document.querySelector('input[name=pinned-post-checkbox]')
+	: null;
```

Comment:
Should we be using `undefined` here instead of `null`? cc @mxdvl (I think undefined is generally preferred but I forget why)

Reply:
I think it needs to be `null` as that is what `querySelector` returns when nothing is found

Reply:
```ts
const pinnedPost: Element | undefined = !isServer && window.document.querySelector('[data-gu-marker=pinned-post]') || undefined;
```
Could work?

Reply:
Why is `undefined` better than `null`? Isn't it that a variable is undefined when it's not been, you know, defined and then in this case one of the possible outcomes is that it becomes `null` which is the value you get when document queries can't find anything.

There used to be a similar thing with `isSignedIn` where `undefined` meant that the check had not completed yet but `false` meant it had completed but the user wasn't signed in. Obviously in this case though the definition is inline so there's never an `undefined` state.

Reply:
I don't think it's a massive deal, though like with everything, there are people with stronger opinions who lean towards or away from using one or the other. 

There were some discussions in Client side around it recently and my interpretation had been that 'undefined' was the generally preferred practice. 

I don't see any material difference between the code that's there and the suggestion I wrote besides whether we have a preferred method as a team/codebase, so it's there if we do!

Reply:
To be honest, I find these kinds of chained ternaries like `const foo = !isApple && isGreen || canBeEaten` hurts my brains 😵‍💫

Reply:
Re the null / undefined wars. I'm honestly pretty clueless what it's all about but we do have several examples of this same pattern in the code (I think this code is based on the `Liveness` file) so my preference is to keep things consistent and maybe revisit this separately?